python/ray/experimental/data/read_api.py

```diff
@@ -103,21 +103,28 @@ def range_arrow(n: int, parallelism: int = 200) -> Dataset[ArrowRow]:
 @PublicAPI(stability="beta")
 def read_datasource(datasource: Datasource[T],
                     parallelism: int = 200,
+                    num_cpus: float = None,
```

Comment:
Please change this to just `remote_args` as in the Dataset .map() definition.

Reply:
Do you mean to combine it with the existing `**read_args`? Or adding an argument `remote_args: Dict[str, Any]`?

Reply:
The latter, exactly as specified in .map().

Reply:
Am I misunderstanding something? in .map, `**remote_args` is a kwargs field. In this function we already have `**read_args` as the kwargs, which would mean they are combined.

Reply:
I see, the issue is you can't have two kwargs fields. How about we make this a normal dict then (not kwargs).